store/store.go

```diff
@@ -120,54 +125,48 @@ func (s *BoltStore) Close() error {
 	return s.db.Close()
 }
 
-// setupDatabase create all buckets in the db.
-// Additionally, it makes sure that the admin user exists.
-func (s *BoltStore) setupDatabase() error {
-	// Create bucket if not exists function
-	var bucketName []byte
+type setup struct {
+	bs  *BoltStore
+	err error
+}
+
+// Create bucket if not exists function
+func (s *setup) update(bucketName []byte) {
+	// update is a no-op in case there was already an error
+	if s.err != nil {
+		return
+	}
 	c := func(tx *bolt.Tx) error {
 		_, err := tx.CreateBucketIfNotExists(bucketName)
 		if err != nil {
 			return fmt.Errorf("create bucket: %s", err)
 		}
 		return nil
 	}
+	s.err = s.bs.db.Update(c)
+}
 
-	// Make sure buckets exist
-	bucketName = userBucket
-	err := s.db.Update(c)
-	if err != nil {
-		return err
-	}
-	bucketName = userPermsBucket
-	err = s.db.Update(c)
-	if err != nil {
-		return err
-	}
-	bucketName = pipelineBucket
-	err = s.db.Update(c)
-	if err != nil {
-		return err
-	}
-	bucketName = createPipelineBucket
-	err = s.db.Update(c)
-	if err != nil {
-		return err
-	}
-	bucketName = pipelineRunBucket
-	err = s.db.Update(c)
-	if err != nil {
-		return err
-	}
-	bucketName = settingsBucket
-	err = s.db.Update(c)
-	if err != nil {
-		return err
+// setupDatabase create all buckets in the db.
+// Additionally, it makes sure that the admin user exists.
+func (s *BoltStore) setupDatabase() error {
+	// Create bucket if not exists function
+	setP := &setup{
+		bs:  s,
+		err: nil,
 	}
-	bucketName = workerBucket
-	err = s.db.Update(c)
-	if err != nil {
-		return err
+
+	// Make sure buckets exist
+	setP.update(userBucket)
```

Comment:
@michelvocks Sorry, I had to refactor this in this batch. It was too repetitive. :)

Reply:
😄 good job!